pyproject.toml

```diff
@@ -26,21 +26,7 @@ requires-python = ">=3.8"
 #
 # For an analysis of this field vs pip's requirements files see:
 # https://packaging.python.org/discussions/install-requires-vs-requirements/
-dependencies = [
-    "arviz>=0.14.0",
-    "graphviz",
-    "ipython!=8.7.0",
-    "matplotlib>=3.5.3",
-    "numpy<1.26.0",
-    "pandas",
-    "patsy",
-    "pymc>=5.0.0",
-    "scikit-learn>=1",
-    "scipy",
-    "seaborn>=0.11.2",
-    "statsmodels",
-    "xarray>=v2022.11.0",
-]
+dependencies = []
```

Comment:
I think the problem comes because you are deleting these dependencies and you are pip installing the package in the CI

Reply:
I think you can try to mimic the way is done in https://github.com/pymc-devs/pymc-experimental/blob/main/.github/workflows/test.yml

Reply:
I'm not that experienced with GitHub workflows. What am I looking for? Are you referring to the `environment-file:` in the `with:` block?

Reply:
Hey, I'm on vacation now, but we should discuss (and update) #281.

As for Conda vs pyproject dependencies, ideally we keep both. Pyproject dependencies define the python-only dependencies that you have. On the other hand, Conda takes care of both Python and non-Python dependencies.

I haven't read carefully any of the details, so I may be missing stuff, but the way I like to handle this is to:
1. Create a Conda environment that satisfies all the dependencies, Python or otherwise
2. `pip install` the project into the Conda environment with the `--no-dependencies` flag so that `pip` installs only the project
3. Run `pip check` to ensure all of the project's Python dependencies are satisfied within the Conda environment. This will trip on the case when you declare a Python dependency in `pyproject.toml` that is missing from your `environment.yaml` file. I don't really know of any tests that go in the other direction, and I don't know that it would even make sense.

Note that this means that your primary source of truth for a project's dependencies should be the `pyproject.toml` file, not the `environment.yaml` file! I think this is how it should be. It enables masochistic "experts" to do an unsupported `pip install` if they are so inclined.

Regarding #281, `conda-lock` can [parse](https://github.com/pymc-labs/CausalPy/blob/b92084fc2f5b1f924a899ad5ec00269f89ef687e/.github/workflows/update-lockfile.yml#L34) the `pyproject.toml` for the requirements and generate a lockfile, so apart from specifying special requirements, we may be able to replace the `environment.yml` with a lockfile.